BUG: pd.Series.interpolate(method='spline') Errort Msg, #10633

springcoil opened the pull request:
Closes #10633

This a first attempt at the error messages mentioned. Not ready to merge yet but I wanted to start the conversation... 
I think I might have the error message wrong - anyone got any ideas.

jreback commented on pandas/core/common.py:
```diff
@@ -1718,6 +1718,9 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
     elif method == 'spline':
+        # GH #10633: first attempt
+        if k is None:
+            raise AssertionError("order needs to be specified, use 1, 2, or 3")
```
use a ``ValueError``

springcoil replied:
Thanks :)
ᐧ

On Thu, Aug 13, 2015 at 12:33 PM, Jeff Reback <notifications@github.com>
wrote:

> In pandas/core/common.py
> <https://github.com/pydata/pandas/pull/10800#discussion_r36960558>:
>
> > @@ -1718,6 +1718,9 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
> >                                      bounds_error=bounds_error)
> >          new_y = terp(new_x)
> >      elif method == 'spline':
> > +        # GH #10633: first attempt
> > +        if k is None:
> > +            raise AssertionError("order needs to be specified, use 1, 2, or 3")
>
> use a ValueError
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/pandas/pull/10800/files#r36960558>.
>



-- 
Peadar Coyle
Skype: springcoilarch
www.twitter.com/springcoil
peadarcoyle.wordpress.com

springcoil commented:
Look like I need to add a test for k - or something or am i wrong?

springcoil commented:
I think that's ready to be merged :)

jreback commented:
needs a test

springcoil commented:
Done :)

jreback commented on pandas/core/common.py:
```diff
@@ -1718,6 +1718,9 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
                                     bounds_error=bounds_error)
         new_y = terp(new_x)
     elif method == 'spline':
+        # GH #10633: first attempt
```
remove the 'first attempt' comment

jreback commented on pandas/core/common.py:
```diff
@@ -1718,6 +1718,9 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
     elif method == 'spline':
+        # GH #10633: first attempt
+        if order is None:
+            raise ValueError("order needs to be specified, use 1, 2, or 3")
```
does scipy raise a nice message when order is ``not None`` but invalid? e.g. 0

jreback commented on pandas/tests/test_generic.py:
```diff
@@ -766,6 +766,14 @@ def test_nan_interpolate(self):
         tm._skip_if_no_scipy()
         result = s.interpolate(method='polynomial', order=1)
         assert_series_equal(result, expected)
+    # GH #10633: first attempt
```
put a blank line in between tests. remove the 'first attempt'

jreback commented on pandas/tests/test_generic.py:
```diff
@@ -766,6 +766,14 @@ def test_nan_interpolate(self):
+        s[np.random.randint(0,9,3)] = np.nan
+        with tm.assertRaises(ValueError):
+            s.interpolate(method='spline')
+        
```
try with order specified but invalid

jreback commented:
pls add a whatsnew note in bug fixes

springcoil commented:
I added a work through etc - maybe it needs an error message (another PR) for order=0 or order>=6

jreback commented on pandas/tests/test_generic.py:
```diff
@@ -1365,6 +1374,17 @@ def test_no_order(self):
+        s = Series(np.arange(10)**2)
+        s[np.random.randint(0,9,3)] = np.nan
+        result1 = s.interpolate(method='spline', order=1)
+        expected1 = s.interpolate(method='spline', order=1)
```
what does this test?

springcoil replied:
I think my test needs some work. You're right.

jreback commented on pandas/core/common.py:
```diff
@@ -1718,6 +1718,9 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
         new_y = terp(new_x)
     elif method == 'spline':
+        # GH #10633
+        if order == None or 0:
```
use ``not order``

springcoil replied:
thanks

jreback commented on pandas/core/common.py:
```diff
@@ -1718,6 +1718,9 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
+        # GH #10633
+        if order == None or 0:
+            raise ValueError("order needs to be specified and greater than 0")
         terp = interpolate.UnivariateSpline(x, y, k=order, **kwargs)
```
add a test that has an order that fails this

springcoil commented:
Updating on the basis of your feedback.

springcoil commented:
I assume this one will be ok for the 0.17 release right?

jreback commented:
ok, pls rebase and squash. ping on green.

Commit:
ENH: pickle support for Period #10439

update legacy_storage for pickles

update pickles/msgpack for 0.16.2

Added tests for ABC Types, Issue #10828

TST: #10822, skip tests on windows for odd error message in to_datetime with unicode

COMPAT:Allow multi-indexes to be written to excel.

(Even though they cannot be read back in.)

Closes #10564

DOC: typo

A few changes in docs

Commit:
ENH: pickle support for Period #10439

update legacy_storage for pickles

update pickles/msgpack for 0.16.2

Added tests for ABC Types, Issue #10828

TST: #10822, skip tests on windows for odd error message in to_datetime with unicode

COMPAT:Allow multi-indexes to be written to excel.

(Even though they cannot be read back in.)

Closes #10564

DOC: typo

ENH: pickle support for Period #10439

update legacy_storage for pickles

update pickles/msgpack for 0.16.2

Added tests for ABC Types, Issue #10828

TST: #10822, skip tests on windows for odd error message in to_datetime with unicode

COMPAT:Allow multi-indexes to be written to excel.

(Even though they cannot be read back in.)

Closes #10564

DOC: typo

A few changes in docs

springcoil committed:
BUG: 10633 and 10800 fix

springcoil committed:
BUG: 10633 - some last errors removed

springcoil committed:
TST: Changes in test

springcoil commented:
Looks like this one got messed up a bit - do you have any suggestions on how to fix this?
I've tried rebasing to upstream - which is the pydata/pandas branch. 
I made the mistake here of not branching correctly from the start.

springcoil commented:
Actually this seems to not be running through travis. I think I pushed to the wrong location. I'll fix it later.

jorisvandenbossche commented:
If you do 

```
git fetch upstream
git rebase -i upstream/master
git push -f origin/master
```

everything should be fine

springcoil commented:
Upstream should be my form right? Or the pandas original one?
ᐧ

On Fri, Aug 21, 2015 at 1:17 PM, Joris Van den Bossche <
notifications@github.com> wrote:

> If you do
>
> git fetch upstream
> git rebase -i upstream/master
> git push -f origin/master
>
> everything should be fine
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/pandas/pull/10800#issuecomment-133374965>.
>



-- 
Peadar Coyle
Skype: springcoilarch
www.twitter.com/springcoil
peadarcoyle.wordpress.com

jorisvandenbossche commented:
upstream should be 'pydata/pandas', origin is 'springcoil/pandas'

You can check this with `git remote -v`

springcoil commented:
Cool thanks for that reminder.
I'll check that after work :)
ᐧ

On Fri, Aug 21, 2015 at 1:23 PM, Joris Van den Bossche <
notifications@github.com> wrote:

> upstream should be 'pydata/pandas', origin is 'springcoil/pandas'
>
> You can check this with git remote -v
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/pandas/pull/10800#issuecomment-133376434>.
>



-- 
Peadar Coyle
Skype: springcoilarch
www.twitter.com/springcoil
peadarcoyle.wordpress.com

springcoil commented:
Thanks @jorisvandenbossche

jreback commented:
replaced by #10880 

normally don't open a new PR as you can simply replace all of the commits. but ok for now.

springcoil commented:
Accidentally done

On Friday, 21 August 2015, Jeff Reback <notifications@github.com> wrote:

> Closed #10800 <https://github.com/pydata/pandas/pull/10800>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/pandas/pull/10800#event-389161182>.
>


-- 
Peadar Coyle
Skype: springcoilarch
www.twitter.com/springcoil
peadarcoyle.wordpress.com
